students/charlie_england/session05/mailroom_part3.py

```diff
@@ -0,0 +1,115 @@
+
+
+def main_menu():
+    '''
+    Main menu for the program.
+    Will loop through and requests input until 3 is entered and then quits.
+    1 will allow an addition of a donator, a new donation and the printing of a thank you letter
+    2 will print a report of the donators and their donations as well as their average donation.
+    3 will send letters to all donors by making .txt files for each
+    4 will quit the program
+    '''
+    switcher = {
+        "1": thank_you,
+        "2": report,
+        "3": make_letters,
+        "4": halt
+    }
+    while True:
+        try:
+            argument = input("""What would you like to do?
+            1: Send a Thank You to a single donor
+            2: Create a Report
+            3: Send letters to all donors
+            4: Quit
+            >>>""").strip()
+            func = switcher.get(argument, "Invalid Result")
+            print(func())
+        except TypeError:
+            print(f"You entered {argument}, please enter 1,2,3 or 4.")
+
+
+def halt():
+    print("Quitting...")
+    sys.exit()
+
+def thank_you():
+    """
+    will ask for a name input or a list.
+    Will print a list of names if list is entered
+    If name is in the dict, will ask for a donation amount, if name is not in dict, will ask if you would like to add a new name and will ask for a donation amount
+    """
+    while True:
+        name_input = input(
+            "Please enter a full name, 'list' for list of names, or back to go back to previous menu \n>>>").strip()
+        if name_input in donors.keys():
+            # Ask for donation amount and append to key in dict
+            donors[name_input][0].append(
+                float(input("Please enter a donation amount: >>> ")))
+            # Call compose thank you function and return breaking the loop
+            return compose_thank_you(name_input)
+        elif name_input == "list":
+            for name in donors.keys():
+                print(name)
+        elif name_input == "back":
+            return "returning to main menu..."
+        else:
+            # if the name_input was not list and was not already in the donors dict, will ask if the user wants to add
+            new_name_decision = input(
+                name_input + " not found, would you like to add a new donator? y/n \n>>>").strip()
+            if new_name_decision.lower() == "y":
+                donors.update(
+                    {name_input: [[int(input("Please enter a donation amount: \n>>>"))],[]]})
+                return compose_thank_you(name_input)
+
+
+def compose_thank_you(name_input):
+    return (
+        """Dear {},\n\n 
+        On behalf of Local Charity we would like to extend our sincerest thanks for your ${:,.2f} donation.\n
+        Without people like you we could not continue blah blah blah\n
+        Again thank you\n
+    Sincerely,\n
+        Local Charity """.format(name_input, int(donors[name_input][0][-1])))
+
+
+def report():
```

Comment:
I see, you're using the second list to keep track of stats. You can ignore my previous comment! Since you will be generating stats every time you run this command, I'm not sure you'll want to keep track of them but that is up to you!